### PR TITLE
Deconvolution2D fix.

### PR DESCRIPTION
**Pull request by lukedeo**
Previously, running the example in [the documentation](https://keras.io/layers/convolutional/#deconvolution2d) resulted in an error for the TF backend. This fix adds the ability to handle `None` in the first dim of the `output_shape` for the TF backend. Added a test that passes for TH, but fails for TF, which under the changes from the PR now passes for both.

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks.
